source/base/bounding_box.cc

```diff
@@ -215,7 +222,8 @@ BoundingBox<spacedim, Number>::side_length(const unsigned int direction) const
 {
   AssertIndexRange(direction, spacedim);
 
-  return boundary_points.second[direction] - boundary_points.first[direction];
+  return std::abs(boundary_points.second[direction] -
+                  boundary_points.first[direction]);
```

Comment:
Isn't it an invariant of the class that the second point is always to the right of the first? I would much rather add an assertion to all of the places where one can set the boundary points to enforce the invariant, than to add the `abs` here.

Reply:
I took it from the point_inside() function, but I thin you are right.